frontend/src/main/java/com/google/tripmeout/frontend/serialization/GsonPlaceVisitTypeAdapter.java

```diff
@@ -0,0 +1,90 @@
+
+import com.google.appengine.repackaged.com.google.api.client.json.Json;
+import com.google.common.base.Strings;
+import com.google.gson.JsonParseException;
+import com.google.gson.TypeAdapter;
+import com.google.gson.stream.JsonReader;
+import com.google.gson.stream.JsonToken;
+import com.google.gson.stream.JsonWriter;
+import com.google.tripmeout.frontend.PlaceVisitModel;
+import com.google.tripmeout.frontend.PlaceVisitModel.UserMark;
+import java.io.IOException;
+
+public class GsonPlaceVisitTypeAdapter extends TypeAdapter<PlaceVisitModel> {
+  private static final String PLACE_VISIT_MODEL_PLACE_ID_JSON_FIELD_NAME = "placeId";
+  private static final String PLACE_VISIT_MODEL_TRIP_ID_JSON_FIELD_NAME = "tripId";
+  private static final String PLACE_VISIT_MODEL_NAME_JSON_FIELD_NAME = "name";
+  private static final String PLACE_VISIT_MODEL_USER_MARK_JSON_FIELD_NAME = "userMark";
+  private static final String PLACE_VISIT_MODEL_LATITUDE_JSON_FIELD_NAME = "latitude";
+  private static final String PLACE_VISIT_MODEL_LONGITUDE_JSON_FIELD_NAME = "longitude";
+
+  @Override
+  public PlaceVisitModel read(JsonReader reader) throws IOException {
+    if (reader.peek() == JsonToken.NULL) {
+      reader.nextNull();
+      return null;
+    }
+    PlaceVisitModel.Builder placeBuilder = PlaceVisitModel.builder();
+    reader.beginObject();
+    while (reader.hasNext()) {
+      JsonToken token = reader.peek();
+      if (token.equals(JsonToken.NAME)) {
+        String name = reader.nextName();
+        switch (name) {
+          case PLACE_VISIT_MODEL_PLACE_ID_JSON_FIELD_NAME:
+            placeBuilder.setPlaceId(reader.nextString());
+            break;
+          case PLACE_VISIT_MODEL_NAME_JSON_FIELD_NAME:
+            placeBuilder.setName(reader.nextString());
+            break;
+          case PLACE_VISIT_MODEL_TRIP_ID_JSON_FIELD_NAME:
+            placeBuilder.setTripId(reader.nextString());
+            break;
+          case PLACE_VISIT_MODEL_USER_MARK_JSON_FIELD_NAME:
+            UserMark userMark = UserMark.valueOf(reader.nextString());
+            placeBuilder.setUserMark(userMark);
+            break;
+          case PLACE_VISIT_MODEL_LATITUDE_JSON_FIELD_NAME:
+            placeBuilder.setLatitude(reader.nextDouble());
+            break;
+          case PLACE_VISIT_MODEL_LONGITUDE_JSON_FIELD_NAME:
+            placeBuilder.setLongitude(reader.nextDouble());
+            break;
+          default:
+            throw new JsonParseException(
+                String.format("Unknown field name %s for type Trip Model", name));
+        }
+      }
+    }
+    reader.endObject();
+    return placeBuilder.build();
+  }
+
+  @Override
+  public void write(JsonWriter writer, PlaceVisitModel place) throws IOException {
+    if (place == null) {
+      writer.nullValue();
+      return;
+    }
+    writer.beginObject();
+    writeUnlessNullOrEmpty(writer, PLACE_VISIT_MODEL_PLACE_ID_JSON_FIELD_NAME, place.placeId());
```

Comment:
Nit: not sure that `writeUnlessNullOrEmpty` is necessary in most of these cases since these are not nullable fields. The values could theoretically be empty strings, but we probably don't expect that in practice. At the end of the day, this is an optimization to save space on the wire if we expect empty values in practice (which I we don't, as far as I can tell).